activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -114,11 +114,15 @@ def sum(identity = nil)
 # just calling the compat method in the first place.
 if Array.instance_methods(false).include?(:sum) && !(%w[a].sum rescue false)
   class Array
-    remove_method :sum
+    alias :orig_sum :sum
 
-    def sum(*args) #:nodoc:
-      # Use Enumerable#sum instead.
-      super
+    def sum(init = nil, &block) #:nodoc:
+      if init.is_a?(Numeric) || self.first.is_a?(Numeric)
```

Comment:
`.first` is not 100% reliable behavior (you could have array `[1, "2"]`. granted it's probably not very common, and `all?` check could be much slower.